lib/modem_info/modem_info_params.c

```diff
@@ -139,53 +139,102 @@ int modem_info_params_get(struct modem_param_info *modem)
 
 	if (IS_ENABLED(CONFIG_MODEM_INFO_ADD_NETWORK)) {
 		ret = modem_data_get(&modem->network.current_band);
-		ret += modem_data_get(&modem->network.sup_band);
-		ret += modem_data_get(&modem->network.ip_address);
-		ret += modem_data_get(&modem->network.ue_mode);
-		ret += modem_data_get(&modem->network.current_operator);
-		ret += modem_data_get(&modem->network.cellid_hex);
-		ret += modem_data_get(&modem->network.area_code);
-		ret += modem_data_get(&modem->network.lte_mode);
-		ret += modem_data_get(&modem->network.nbiot_mode);
-		ret += modem_data_get(&modem->network.gps_mode);
-		ret += modem_data_get(&modem->network.apn);
+		if (ret) {
+			return -EAGAIN;
+		}
+		ret = modem_data_get(&modem->network.sup_band);
+		if (ret) {
+			return -EAGAIN;
+		}
+		ret = modem_data_get(&modem->network.ip_address);
+		if (ret) {
+			return -EAGAIN;
+		}
+		ret = modem_data_get(&modem->network.ue_mode);
+		if (ret) {
+			return -EAGAIN;
+		}
+		ret = modem_data_get(&modem->network.current_operator);
+		if (ret) {
+			return -EAGAIN;
+		}
+		ret = modem_data_get(&modem->network.cellid_hex);
+		if (ret) {
+			return -EAGAIN;
+		}
+		ret = modem_data_get(&modem->network.area_code);
+		if (ret) {
+			return -EAGAIN;
+		}
+		ret = modem_data_get(&modem->network.lte_mode);
+		if (ret) {
+			return -EAGAIN;
+		}
+		ret = modem_data_get(&modem->network.nbiot_mode);
+		if (ret) {
+			return -EAGAIN;
+		}
+		ret = modem_data_get(&modem->network.gps_mode);
+		if (ret) {
+			return -EAGAIN;
+		}
+		ret = modem_data_get(&modem->network.apn);
+		if (ret) {
+			return -EAGAIN;
+		}
 
 		if (IS_ENABLED(CONFIG_MODEM_INFO_ADD_DATE_TIME)) {
-			ret += modem_data_get(&modem->network.date_time);
+			ret = modem_data_get(&modem->network.date_time);
+			if (ret) {
+				LOG_ERR("Could not get time, error: %d", ret);
+				/* non-critical error: continue */
+			}
 		}
 
-		ret += mcc_mnc_parse(&modem->network.current_operator,
+		ret = mcc_mnc_parse(&modem->network.current_operator,
 				&modem->network.mcc,
 				&modem->network.mnc);
-		ret += cellid_to_dec(&modem->network.cellid_hex,
+		if (ret) {
+			return -EAGAIN;
```

Comment:
Why return `-EAGAIN` instead of the original error code? (This comment goes for all places where we return `-EAGAIN` instead of the original error).

Reply:
fixed